### PR TITLE
Added suitable null checks to the extension method arguments

### PR DESCRIPTION
**Pull request by d-saravanan**
Summary of the changes (Less than 80 chars)
 - Added suitable null checks to the extension method arguments

Addresses #316

### CODE REVIEW

**Comment by d-saravanan** [comment]
Can anyone trigger a new build on this, because the build in my repository is passing and only here its failing

File: src/Microsoft.Extensions.Caching.Abstractions/MemoryCacheExtensions.cs [hunk 1]
```diff
@@ -45,6 +65,10 @@ public static TItem Set<TItem>(this IMemoryCache cache, object key, TItem value)
 
         public static TItem Set<TItem>(this IMemoryCache cache, object key, TItem value, DateTimeOffset absoluteExpiration)
         {
+            if (key == null)
+            {
+                throw new System.ArgumentNullException(nameof(key));
```
Comment (Tratcher) [inline src/Microsoft.Extensions.Caching.Abstractions/MemoryCacheExtensions.cs]: Add a using for System.

File: src/Microsoft.Extensions.Caching.Abstractions/MemoryCacheExtensions.cs [hunk 2]
```diff
@@ -12,18 +12,33 @@ public static class CacheExtensions
 
         public static object Get(this IMemoryCache cache, object key)
         {
+            if (key == null)
```
Comment (jkotalik) [inline src/Microsoft.Extensions.Caching.Abstractions/MemoryCacheExtensions.cs]: Can we add a few tests to see that a ArgumentNullException is thrown. See: https://github.com/aspnet/Caching/blob/dev/test/Microsoft.Extensions.Caching.Redis.Tests/RedisCacheSetAndRemoveTests.cs#L93-L97 for a good example.

Comment (d-saravanan) [reply src/Microsoft.Extensions.Caching.Abstractions/MemoryCacheExtensions.cs]: I have added 6 test cases to test all the different scenario's and have also refactored the MemoryCacheExtensions.cs [CacheExtensions] to remove the redundant validations made in the earlier check-in. Please review and share your review comments.

File: src/Microsoft.Extensions.Caching.Abstractions/MemoryCacheExtensions.cs (hunk 1 above)
Comment (d-saravanan) [reply src/Microsoft.Extensions.Caching.Abstractions/MemoryCacheExtensions.cs]: I have updated the code and sent a new pull-request and raised a review.

File: src/Microsoft.Extensions.Caching.Abstractions/MemoryCacheExtensions.cs [hunk 3]
```diff
@@ -119,5 +128,15 @@ public static TItem GetOrCreate<TItem>(this IMemoryCache cache, object key, Func
 
             return (TItem)result;
         }
+
+        #region Private Members
```
Comment (Tratcher) [inline src/Microsoft.Extensions.Caching.Abstractions/MemoryCacheExtensions.cs]: No regions please.

**Comment by davidfowl** [comment]
I hope we don't double null check in the default scenario.

**Comment by d-saravanan** [comment]
@davidfowl I have ensured there are no double null checks. However, if we can move the null check to MemoryCache.cs [CreateEntry method], we can remove the checks that are present in the MemoryCacheExtensions.cs [CacheExtensions class]. Please let me know if I should move the checks inside MemoryCache.

File: src/Microsoft.Extensions.Caching.Abstractions/MemoryCacheExtensions.cs (hunk 3 above)
Comment (d-saravanan) [reply src/Microsoft.Extensions.Caching.Abstractions/MemoryCacheExtensions.cs]: @Tratcher : For my learning, can you please let me know why not to use regions. I have been using them in my source code. Is that not a good practice. Should I remove them and raise a new pull-request.

Comment (Tratcher) [reply src/Microsoft.Extensions.Caching.Abstractions/MemoryCacheExtensions.cs]: It's only a style issue, we don't use them in most cases, especially not for a single method.

**Comment by jkotalik** [comment]
I think we can just add a null check here: https://github.com/aspnet/Caching/blob/dev/src/Microsoft.Extensions.Caching.Memory/MemoryCache.cs#L83. This would simplify the check to one place and reduce redundancy.

**Comment by d-saravanan** [comment]
@jkotalik I had earlier posted in the reply David Fowler. However, I will update and initiate a new pull-request.

**Comment by d-saravanan** [comment]
@jkotalik I have updated and initiated a new pull-request. Kindly review the same and share your comments if any.

**Pull request merged**